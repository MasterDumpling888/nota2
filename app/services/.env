GOOGLE_GENAI_API_KEY=AIzaSyDKoSfHEB0oWlEdXNH-BmeFsZQuDoT2Dqs