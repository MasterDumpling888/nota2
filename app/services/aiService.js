import axios from 'axios';

const BACKEND_API_URL = 'http://localhost:3000';

export const summarizeNote = async (content) => {
  try {
    const response = await axios.post(`${BACKEND_API_URL}/summarize`, { content });
    return response.data.summary;
  } catch (error) {
    console.error('Summarization error:', error.response?.data || error.message);
    throw new Error('Failed to connect to the summarization service');
  }
};

export const generateQuestionsAndAnswers = async (content) => {
    try {
     
      const response = await axios.post(`${BACKEND_API_URL}/generate-qa`, { content });
      
      return response.data.qaPairs;
    } catch (error) {
      console.error('Q&A generation error:', error.response?.data || error.message);
      throw new Error('Failed to generate questions and answers');
    }
  };
  
const testAI = async () => {
const content = "The Industrial Revolution, which began in the late 18th century, marked a major turning point in history. Almost every aspect of daily life was influenced in some way. The introduction of new machines, such as the steam engine, revolutionized manufacturing and transportation. Factories emerged, leading to urbanization as people moved to cities in search of work. While the Industrial Revolution brought about significant economic growth and technological advancements, it also had negative consequences, including poor working conditions, child labor, and environmental degradation. Over time, labor laws and reforms were introduced to address these issues, but the impact of this era is still felt today.";
try {
    const summary = await summarizeNote(content);
    console.log('Summary:', summary);

    const qaPairs = await generateQuestionsAndAnswers(content);
    console.log('Questions and Answers:');
    qaPairs.forEach((pair, index) => {
    console.log(`\nQ${index+1}: ${pair.question}`);
    console.log(`A${index+1}: ${pair.answer}`);
    });
} catch (error) {
    console.error('Test failed:', error);
}
};

// Run the test
testAI();