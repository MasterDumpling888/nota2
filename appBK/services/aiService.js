import { GoogleGenerativeAI } from '@google/generative-ai';

// Initialize Gemini
const genAI = new GoogleGenerativeAI('AIzaSyDKoSfHEB0oWlEdXNH-BmeFsZQuDoT2Dqs');
const model = genAI.getGenerativeModel({ model: 'gemini-1.5-flash' });

export const summarizeNote = async (content) => {
  try {
    const prompt = `Summarize the following text in 2-3 sentences:\n\n${content}`;
    const result = await model.generateContent(prompt);
    return result.response.text();
  } catch (error) {
    console.error('Failed to summarize text', error);
    throw new Error('Failed to summarize text');
  }
};

export const generateReviewQuestions = async (content) => {
  try {
    const prompt = `Generate 3 questions and answers based on the following text. Format the response as a JSON array where each object contains a "question" and "answer" field:\n\n${content}`;
    const result = await model.generateContent(prompt);
    const responseText = result.response.text();

    // Parse the response as JSON
    let qaPairs;
    try {
      // Try to extract JSON array from the response
      const jsonMatch = responseText.match(/\[[\s\S]*\]/);
      if (jsonMatch) {
        qaPairs = JSON.parse(jsonMatch[0]);
      } else {
        throw new Error('Invalid JSON format in response');
      }
    } catch (parseError) {
      console.error('Error parsing JSON response:', parseError);
      console.log('Raw response:', responseText);

      // Fallback: Try to format response manually
      // This is a basic fallback in case the model doesn't return proper JSON
      qaPairs = [];
      const sections = responseText.split(/\d+\.\s*Question:/i);
      sections.shift(); // Remove first empty section if any

      for (const section of sections) {
        const questionMatch = section.match(/^(.*?)(?:Answer:)/is);
        const answerMatch = section.match(/Answer:(.*?)(?:\d+\.\s*Question:|$)/is);

        if (questionMatch && answerMatch) {
          qaPairs.push({
            question: questionMatch[1].trim(),
            answer: answerMatch[1].trim()
          });
        }
      }
    }

    return qaPairs;
  } catch (error) {
    console.error('Failed to generate questions and answers', error);
    throw new Error('Failed to generate questions and answers');
  }
};
