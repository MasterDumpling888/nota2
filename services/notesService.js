import { db, auth } from '../firebaseConfig';
import { collection, addDoc, updateDoc, deleteDoc, getDocs, doc, getDoc } from 'firebase/firestore';

const getUserNotesCollection = () => {
  const user = auth.currentUser;
  if (!user) throw new Error('User not authenticated');
  return collection(db, 'users', user.uid, 'notes');
};

const getUserFoldersCollection = () => {
  const user = auth.currentUser;
  if (!user) throw new Error('User not authenticated');
  return collection(db, 'users', user.uid, 'folders');
};

const getUserTagsCollection = () => {
  const user = auth.currentUser;
  if (!user) throw new Error('User not authenticated');
  return collection(db, 'users', user.uid, 'tags');
};

export const addNote = async (note) => {
  try {
    const notesCollection = getUserNotesCollection();
    const docRef = await addDoc(notesCollection, note);
    return docRef.id;
  } catch (error) {
    console.error("Error adding note: ", error);
    throw error;
  }
};

export const updateNote = async (id, updatedNote) => {
  try {
    const notesCollection = getUserNotesCollection();
    const noteDoc = doc(notesCollection, id);
    await updateDoc(noteDoc, updatedNote);
  } catch (error) {
    console.error("Error updating note: ", error);
    throw error;
  }
};

export const deleteNote = async (id) => {
  try {
    const notesCollection = getUserNotesCollection();
    const noteDoc = doc(notesCollection, id);
    await deleteDoc(noteDoc);
  } catch (error) {
    console.error("Error deleting note: ", error);
    throw error;
  }
};

export const fetchNotes = async () => {
  try {
    const notesCollection = getUserNotesCollection();
    const querySnapshot = await getDocs(notesCollection);
    const notesList = querySnapshot.docs.map(doc => ({ id: doc.id, ...doc.data() }));
    return notesList;
  } catch (error) {
    console.error("Error fetching notes: ", error);
    throw error;
  }
};

export const getNote = async (id) => {
  try {
    const notesCollection = getUserNotesCollection();
    const noteDoc = doc(notesCollection, id);
    const noteSnapshot = await getDoc(noteDoc);
    if (noteSnapshot.exists()) {
      return { id: noteSnapshot.id, ...noteSnapshot.data() };
    } else {
      throw new Error('Note not found');
    }
  } catch (error) {
    console.error("Error getting note: ", error);
    throw error;
  }
};

export const addFolder = async (folder) => {
  try {
    const foldersCollection = getUserFoldersCollection();
    const docRef = await addDoc(foldersCollection, folder);
    return docRef.id;
  } catch (error) {
    console.error("Error adding folder: ", error);
    throw error;
  }
};

export const addTag = async (tag) => {
  try {
    const tagsCollection = getUserTagsCollection();
    const docRef = await addDoc(tagsCollection, tag);
    return docRef.id;
  } catch (error) {
    console.error("Error adding tag: ", error);
    throw error;
  }
};

export const fetchFolders = async () => {
  try {
    const foldersCollection = getUserFoldersCollection();
    const querySnapshot = await getDocs(foldersCollection);
    const foldersList = querySnapshot.docs.map(doc => ({ id: doc.id, ...doc.data() }));
    return foldersList;
  } catch (error) {
    console.error("Error fetching folders: ", error);
    throw error;
  }
};

export const fetchTags = async () => {
  try {
    const tagsCollection = getUserTagsCollection();
    const querySnapshot = await getDocs(tagsCollection);
    const tagsList = querySnapshot.docs.map(doc => ({ id: doc.id, ...doc.data() }));
    return tagsList;
  } catch (error) {
    console.error("Error fetching tags: ", error);
    throw error;
  }
};
