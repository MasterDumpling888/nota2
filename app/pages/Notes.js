import React, { useState, useEffect } from 'react';
import { ScrollView, View, Text, TouchableOpacity, TextInput, StyleSheet, Animated } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { fetchNotes, addFolder, addTag, fetchFolders, fetchTags } from '../services/notesService';
import PageBox from '../components/PageBox';
import Icon from 'react-native-vector-icons/Ionicons';
import { getFontSize } from '../responsiveFont';

const Notes = ({ navigation }) => {
  const [notes, setNotes] = useState([]);
  const [folders, setFolders] = useState([]);
  const [tags, setTags] = useState([]);
  const [newFolder, setNewFolder] = useState('');
  const [newTag, setNewTag] = useState('');
  const [fabExtended, setFabExtended] = useState(false);
  const [animation] = useState(new Animated.Value(0));
  const [showFolderInput, setShowFolderInput] = useState(false);
  const [showTagInput, setShowTagInput] = useState(false);
  const [displayMode, setDisplayMode] = useState('notes');

  useEffect(() => {
    const loadNotes = async () => {
      try {
        const notesList = await fetchNotes();
        setNotes(notesList);
      } catch (error) {
        console.error('Failed to load notes', error);
      }
    };

    const loadFolders = async () => {
      try {
        const foldersList = await fetchFolders();
        setFolders(foldersList);
      } catch (error) {
        console.error('Failed to load folders', error);
      }
    };

    const loadTags = async () => {
      try {
        const tagsList = await fetchTags();
        setTags(tagsList);
      } catch (error) {
        console.error('Failed to load tags', error);
      }
    };

    loadNotes();
    loadFolders();
    loadTags();
  }, []);

  const handleAddFolder = async () => {
    try {
      await addFolder({ name: newFolder });
      setNewFolder('');
      const foldersList = await fetchFolders();
      setFolders(foldersList);
      setShowFolderInput(false);
    } catch (error) {
      console.error('Failed to add folder', error);
    }
  };

  const handleAddTag = async () => {
    try {
      await addTag({ name: newTag });
      setNewTag('');
      const tagsList = await fetchTags();
      setTags(tagsList);
      setShowTagInput(false);
    } catch (error) {
      console.error('Failed to add Tag', error);
    }
  };

  const toggleFab = () => {
    setFabExtended(!fabExtended);
    Animated.timing(animation, {
      toValue: fabExtended ? 0 : 1,
      duration: 300,
      useNativeDriver: true,
    }).start();
  };

  const fabOptionStyle = (index) => ({
    transform: [
      {
        translateX: animation.interpolate({
          inputRange: [0, 1],
          outputRange: [0, -1 * (index)],
        }),
      },
      {
        scale: animation.interpolate({
          inputRange: [0, 1],
          outputRange: [0, 1],
        }),
      },
    ],
    opacity: animation.interpolate({
      inputRange: [0, 1],
      outputRange: [0, 1],
    }),
  });

  const renderNotesByFolder = (folderId) => {
    return notes
      .filter((note) => note.folder === folderId)
      .map((note, index) => (
        <TouchableOpacity
          key={index}
          style={styles.noteContainer}
          onPress={() => navigation.navigate('NoteEditor', { noteId: note.id })}
        >
          <Icon name="document-text-outline" size={24} color="black" />
          <Text style={styles.noteTitle}>{note.title}</Text>
          {/* <Text style={styles.noteContent}>{note.content}</Text> */}
        </TouchableOpacity>
      ));
  };

  const renderNotesByTag = (TagId) => {
    return notes
      .filter((note) => note.tag === TagId)
      .map((note, index) => (
        <TouchableOpacity
          key={index}
          style={styles.noteContainer}
          onPress={() => navigation.navigate('NoteEditor', { noteId: note.id })}
        >
          <Icon name="document-text-outline" size={24} color="black" />
          <Text style={styles.noteTitle}>{note.title}</Text>
          {/* <Text style={styles.noteContent}>{note.content}</Text> */}
        </TouchableOpacity>
      ));
  };

  const renderContent = () => {
    switch (displayMode) {
      case 'notes':
        return notes.map((note, index) => (
          <TouchableOpacity
            key={index}
            style={styles.noteContainer}
            onPress={() => navigation.navigate('NoteEditor', { noteId: note.id })}
          >
            <Icon name="document-text-outline" size={24} color="black" />
            <Text style={styles.noteTitle}>{note.title}</Text>
            <Text style={styles.noteContent}>{note.content}</Text>
          </TouchableOpacity>
        ));
      case 'folders':
        return folders.map((folder, index) => (
          <View key={index} style={styles.folderContainer}>
            <Icon name="folder-outline" size={24} color="black" />
            <Text style={styles.folderTitle}>{folder.name}</Text>
            {renderNotesByFolder(folder.id)}
          </View>
        ));
      case 'tags':
        return tags.map((tag, index) => (
          <View key={index} style={styles.tagContainer}>
            <Icon name="pricetag-outline" size={24} color="black" />
            <Text style={styles.tagTitle}>{tag.name}</Text>
            {renderNotesByTag(tag.id)}
          </View>
        ));
      default:
        return null;
    }
  };

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <PageBox title="Your Notes" onClose={() => navigation.goBack()}>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            style={[styles.displayButton, displayMode === 'notes' && styles.activeButton]}
            onPress={() => setDisplayMode('notes')}
          >
            <Icon name="document-text-outline" size={24} color="black" />
            <Text style={styles.buttonText}>Notes</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={[styles.displayButton, displayMode === 'folders' && styles.activeButton]}
            onPress={() => setDisplayMode('folders')}
          >
            <Icon name="folder-outline" size={24} color="black" />
            <Text style={styles.buttonText}>Folders</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={[styles.displayButton, displayMode === 'tags' && styles.activeButton]}
            onPress={() => setDisplayMode('tags')}
          >
            <Icon name="pricetag-outline" size={24} color="black" />
            <Text style={styles.buttonText}>Tags</Text>
          </TouchableOpacity>
        </View>
        <ScrollView>
          {renderContent()}
        </ScrollView>
        <View style={styles.fabContainer}>
          {fabExtended && (
            <Animated.View style={fabOptionStyle(0)}>
              <TouchableOpacity
                onPress={() => navigation.navigate('NoteEditor')}
                style={styles.fabOption}
              >
                <Icon name="document-text-outline" size={24} color="black" />
                <Text style={styles.fabOptionText}>Note</Text>
              </TouchableOpacity>
            </Animated.View>
          )}
          {fabExtended && (
            <Animated.View style={fabOptionStyle(1)}>
              <TouchableOpacity
                onPress={() => {
                  setShowFolderInput(true);
                  setShowTagInput(false);
                }}
                style={styles.fabOption}
              >
                <Icon name="folder-outline" size={24} color="black" />
                <Text style={styles.fabOptionText}>Folder</Text>
              </TouchableOpacity>
            </Animated.View>
          )}
          {fabExtended && (
            <Animated.View style={fabOptionStyle(2)}>
              <TouchableOpacity
                onPress={() => {
                  setShowTagInput(true);
                  setShowFolderInput(false);
                }}
                style={styles.fabOption}
              >
                <Icon name="pricetag-outline" size={24} color="black" />
                <Text style={styles.fabOptionText}>Tag</Text>
              </TouchableOpacity>
            </Animated.View>
          )}
          <TouchableOpacity
            style={styles.fab}
            onPress={() => {
              toggleFab();
              setShowFolderInput(false);
              setShowTagInput(false);
            }}
          >
            <Icon name={fabExtended ? "close" : "add"} size={35} color="black" />
          </TouchableOpacity>
        </View>
        {showFolderInput && (
          <View style={styles.inputContainer}>
            <TextInput
              style={styles.input}
              placeholder="New Folder"
              value={newFolder}
              onChangeText={setNewFolder}
            />
            <TouchableOpacity style={styles.saveButton} onPress={handleAddFolder}>
              <Text style={styles.saveButtonText}>Add Folder</Text>
            </TouchableOpacity>
          </View>
        )}
        {showTagInput && (
          <View style={styles.inputContainer}>
            <TextInput
              style={styles.input}
              placeholder="New Tag"
              value={newTag}
              onChangeText={setNewTag}
            />
            <TouchableOpacity style={styles.saveButton} onPress={handleAddTag}>
              <Text style={styles.saveButtonText}>Add Tag</Text>
            </TouchableOpacity>
          </View>
        )}
      </PageBox>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  noteContainer: {
    padding: 16,
    borderBottomWidth: 1,
    borderBottomColor: '#32FB0A',
  },
  noteTitle: {
    fontSize: getFontSize(18),
    fontWeight: 'bold',
    marginBottom: 8,
  },
  noteContent: {
    fontSize: getFontSize(14),
  },
  folderContainer: {
    padding: 16,
    borderWidth: 1,
    borderColor: '#32FB0A',
    borderRadius: 10,
    marginTop: 10,
  },
  folderTitle: {
    fontSize: getFontSize(20),
    fontWeight: 'bold',
    marginBottom: 8,
    borderBottomWidth: 1,
    borderBottomColor: '#32FB0A'
  },
  tagContainer: {
    padding: 16,
    borderWidth: 1,
    borderColor: '#32FB0A',
    borderRadius: 10,
    marginTop: 10
  },
  tagTitle: {
    fontSize: getFontSize(20),
    fontWeight: 'bold',
    marginBottom: 8,
    borderBottomWidth: 1,
    borderBottomColor: '#32FB0A'
  },
  inputContainer: {
    marginBottom: 16,
    padding: 16,
  },
  input: {
    borderWidth: 1,
    borderColor: '#32FB0A',
    borderRadius: 5,
    padding: 8,
    marginTop: 8,
    fontSize: getFontSize(16),
  },
  saveButton: {
    backgroundColor: '#32FB0A',
    borderRadius: 5,
    padding: 10,
    marginTop: 8,
    alignItems: 'center',
  },
  saveButtonText: {
    color: 'black',
    fontSize: getFontSize(16),
    fontWeight: 'bold',
  },
  fabContainer: {
    justifyContent: 'flex-end',
    alignItems: 'center',
    backgroundColor: 'transparent',
    paddingTop: 16,
    flexDirection: 'row',
    gap: 45
  },
  fab: {
    backgroundColor: '#32FB0A',
    borderRadius: 50,
    width: 65,
    height: 65,
    justifyContent: 'center',
    alignItems: 'center',
  },
  fabOption: {
    backgroundColor: '#32FB0A',
    borderRadius: 50,
    width: 65,
    height: 65,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    gap: 4,
  },
  fabOptionText: {
    color: 'black',
    fontSize: getFontSize(14),
    fontWeight: 'bold',
    textAlign: 'center'
  },
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginBottom: 16,
  },
  displayButton: {
    padding: 10,
    borderRadius: 50,
    borderWidth: 1,
    borderColor: '#32FB0A',
    flexDirection: 'row',
    alignItems: 'center',
    gap: 8,
  },
  activeButton: {
    backgroundColor: '#32FB0A',
  },
  buttonText: {
    color: 'black',
    fontSize: getFontSize(15),
    fontWeight: 'bold',
  },
});

export default Notes;