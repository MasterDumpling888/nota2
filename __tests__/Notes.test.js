import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import Notes from '../app/pages/Notes';
import { fetchNotes, fetchFolders, fetchTags, addFolder, addTag } from '../app/services/notesService';

// Mock dependencies
jest.mock('../app/services/notesService', () => ({
    fetchNotes: jest.fn(),
    fetchFolders: jest.fn(),
    fetchTags: jest.fn(),
    addFolder: jest.fn(),
    addTag: jest.fn(),
}));

describe('Notes Component', () => {
    const mockNavigation = {
        navigate: jest.fn(),
        goBack: jest.fn()
    };
    
    beforeEach(() => {
        jest.clearAllMocks();
        
        // Mock data
        fetchNotes.mockResolvedValue([
            { id: '1', title: 'Test Note 1', summary: 'Summary 1', folder: '1', tag: '1' },
            { id: '2', title: 'Test Note 2', summary: 'Summary 2', folder: '2', tag: '2' },
        ]);
        
        fetchFolders.mockResolvedValue([
            { id: '1', name: 'Folder 1' },
            { id: '2', name: 'Folder 2' },
        ]);
        
        fetchTags.mockResolvedValue([
            { id: '1', name: 'Tag 1' },
            { id: '2', name: 'Tag 2' },
        ]);
    });

    test('fetches data on component mount', async () => {
        render(<Notes navigation={global.mockNavigation} />);
        
        await waitFor(() => {
            expect(fetchNotes).toHaveBeenCalledTimes(1);
            expect(fetchFolders).toHaveBeenCalledTimes(1);
            expect(fetchTags).toHaveBeenCalledTimes(1);
        });
    });

    test('renders display mode buttons correctly', () => {
        const { getByText } = render(<Notes navigation={global.mockNavigation} />);
        expect(getByText('Notes')).toBeTruthy();
        expect(getByText('Folders')).toBeTruthy();
        expect(getByText('Tags')).toBeTruthy();
    });
    
    test('switches between display modes', async () => {
        const { getByText } = render(<Notes navigation={global.mockNavigation} />);
        
        // Initially in notes mode
        fireEvent.press(getByText('Folders'));
        // Now in folders mode
        
        fireEvent.press(getByText('Tags'));
        // Now in tags mode
        
        fireEvent.press(getByText('Notes'));
        // Back to notes mode
    });
    
    test('navigates to note editor when a note is pressed', async () => {
        const { findByText } = render(<Notes navigation={global.mockNavigation} />);
        
        // Wait for notes to be rendered
        const noteTitle = await findByText('Test Note 1');
        fireEvent.press(noteTitle);
        
        expect(global.mockNavigation.navigate).toHaveBeenCalledWith('NoteEditor', { noteId: '1' });
    });

    // Snapshot testing
    test('Notes snapshot test', () => {
        const { toJSON } = render(<Notes navigation={global.mockNavigation} />);
        expect(toJSON()).toMatchSnapshot();
    });
});
