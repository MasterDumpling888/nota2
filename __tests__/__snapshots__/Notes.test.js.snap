// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes Component Notes snapshot test 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <PageBox
    onClose={[Function]}
    title="Your Notes"
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginBottom": 16,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#33FD0Aa3",
            "borderColor": "#33FD0A",
            "borderRadius": 50,
            "borderWidth": 1,
            "color": "black",
            "flexDirection": "row",
            "gap": 8,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Icon
          color="#33FD0A"
          name="document-text-outline"
          size={24}
        />
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Raleway-SemiBold",
              "fontSize": 15,
            }
          }
        >
          Notes
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#33FD0A",
            "borderRadius": 50,
            "borderWidth": 1,
            "flexDirection": "row",
            "gap": 8,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Icon
          color="#33FD0A"
          name="folder-outline"
          size={24}
        />
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Raleway-SemiBold",
              "fontSize": 15,
            }
          }
        >
          Folders
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#33FD0A",
            "borderRadius": 50,
            "borderWidth": 1,
            "flexDirection": "row",
            "gap": 8,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Icon
          color="#33FD0A"
          name="pricetag-outline"
          size={24}
        />
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Raleway-SemiBold",
              "fontSize": 15,
            }
          }
        >
          Tags
        </Text>
      </View>
    </View>
    <RCTScrollView
      showsVerticalScrollIndicator={false}
    >
      <View />
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "gap": 78.75,
          "justifyContent": "flex-end",
          "paddingTop": 16,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#33FD0A",
            "borderRadius": 56.25,
            "height": 112.5,
            "justifyContent": "center",
            "opacity": 1,
            "width": 112.5,
          }
        }
      >
        <Icon
          color="black"
          name="add"
          size={35}
        />
      </View>
    </View>
  </PageBox>
</RNCSafeAreaView>
`;
